.github/workflows/nightly-release.yaml

```diff
@@ -0,0 +1,76 @@
+name: Nightly Release
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *' ## Every day at midnight UTC
+
+permissions:
+  contents: read
+
+jobs:
+  checkout:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          fetch-depth: 0
+
+  test:
+    runs-on: ubuntu-latest
+    needs: checkout
+    steps:
+      - name: Run tests
+        uses: ./.github/workflows/release-tests.yaml
+
+  push:
+    runs-on: ubuntu-latest
+    environment: release
+    needs: test
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          fetch-depth: 0
+
+      - name: Setup golang
+        uses: ./.github/actions/golang
+
+      - name: Install tools
+        uses: ./.github/actions/install-tools
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: build-artifacts
+          path: build/
+
+      - name: Create tag
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            github.rest.git.createRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: 'refs/tags/nightly',
+              sha: context.sha
+            }).catch(err => {
+              if (err.status !== 422) throw err;
```

Comment:
maybe a comment here that 422 is not processable, i'm guessing because of the ref .... and therefore doing the update in that case?

Reply:
Exactly, if the ref already exists it will fail, so we need to call update, I'll add a comment